MOBILE-9220 added newline to tcp messages, added tests

ezhidblr opened the pull request.

ezhidblr committed:
MOBILE-9220 added newline to tcp messages, added tests

dave-shawley commented on sprockets/mixins/metrics/statsd.py:
```diff
@@ -174,6 +175,12 @@ def send(self, path, value, metric_type):
+            return msg
+
+        return msg + "\n"
+
```
Just make the format an attribute instead of adding another function call and string concatenation to each metric send.  You can set the format string (eg, `'{path}:{value}|{type}'` or `'{path}:{value}|{type}\n'`) in the [initializer](https://github.com/ezhidblr/sprockets.mixins.metrics/blob/6fd2cf40352517839b854b627f4b04c3a6551d19/sprockets/mixins/metrics/statsd.py#L124-L131) and change the send code to:
```python
msg = self._payload_format.format(
   path=self._build_path(path, metric_type),
   value=value, type=metric_type)
```

dave-shawley reviewed (changes requested):
Just make the format an attribute instead of adding another function call and string concatenation to each metric send.  You can set the format string (eg, `'{path}:{value}|{type}'` or `'{path}:{value}|{type}\n'`) in the [initializer](https://github.com/ezhidblr/sprockets.mixins.metrics/blob/6fd2cf40352517839b854b627f4b04c3a6551d19/sprockets/mixins/metrics/statsd.py#L124-L131) and change the send code to:
```python
msg = self._payload_format.format(
   path=self._build_path(path, metric_type),
   value=value, type=metric_type)
```

ezhidblr committed:
Set format strings ass attributes, update tests

ezhidblr committed:
Fakeserver check for \n in tcp stream, const name changed to TCP_PATTERN

ezhidblr committed:
Add tcp & udp tests for data sent

ezhidblr committed:
replace list dereferencing with itertools.chain for py2

ezhidblr committed:
import correct patch for py2 compatibility

bkorty commented:
+1

ezhidblr committed:
- add non-blocking sleep to reconnect
- catch iostream.StreamClosedError
- short circuit write on a closed tcp stream
- flake8 formatting

ezhidblr committed:
Added statsd tests for 100% coverage, upgraded coverage pkg and conf

ezhidblr committed:
Merge branch 'master' into MOBILE-9146-add-statsd-tcp
